### DISCUSSION TITLE
DateEdit Behavior

### DISCUSSION THREAD

**Discussion by mseittenranta** [Megabit/Blazorise#2900] [Q&A] [answered]
The DateEdit component behaves different on the documentation page as to the demo site. 
I'd like to use the behavior I see in the documentation, with no seconds in the time and not blanking out the date and time when a zero is typed in to a date time element.  
Default behavior is that users see the date as, for example, 01-Sep-2021 and want to change to 02-Sep-2021 so they type in a 0 and the date is now 01-Jan-0001 (or dd-----yyyy if DateTime?), however on the Documentation page it shows as 00-Sep-2021, ready for the user to type in the next digit, a 2 in this case.
How do I make this happen?

**Comment by stsrki** [comment, marked as answer]
The DateEdit example in the documentation is a little different than it should be. The reason is mostly historically. Current documentation is not running on Blazor but on Jekyll, because 3 years ago that was the best option to make the documentation SEO friendly. That's why they don't run the same. We have already created new documentation that we soon plan to publish and that is completely done on Blazorise this time. See it here https://preview.blazorise.com/docs/components/date

---

Also, regarding the DateEdit behavior. It is created around the native HTML date input, meaning it renders as `<input type="date">`, and since `type="date"` is used it will be in full control by the browser and system settings. There is not much we can do to make it behave differently. For example your system show format `dd-mmm-yyyy`, and my shows `dd-MM-yyyy`. When I try to type 0 and then 2, for me it is working and it fills as 02.

Because all of this and because we really can't have control over browser, we have introduced new component DatePicker, that has much more option regarding, formating, input mask, etc. You can try it at https://preview.blazorise.com/docs/components/date-picker

**Reply by mseittenranta** [reply]
Thanks for that, I'll see if the DatePicker gives me what I'm looking for.
